integrations/website-indexer/src/index.ts

```diff
@@ -0,0 +1,277 @@
+import { z, RuntimeError } from '@botpress/sdk'
+import axios from 'axios'
+import _ from 'lodash'
+import parseRobots from 'robots-parser'
+import Url from 'url'
+// import xml2json from 'xml2json'
+import xml2js from 'xml2js'
+import { Scraper } from './scraper'
+import * as bp from '.botpress'
+
+export default new bp.Integration({
+  register: async () => {
+    /**
+     * This is called when a bot installs the integration.
+     * You should use this handler to instanciate ressources in the external service and ensure that the configuration is valid.
+     */
+    throw new RuntimeError('Invalid configuration') // replace this with your own validation logic
+  },
+  unregister: async () => {
+    /**
+     * This is called when a bot removes the integration.
+     * You should use this handler to instanciate ressources in the external service and ensure that the configuration is valid.
+     */
+    throw new RuntimeError('Invalid configuration') // replace this with your own validation logic
+  },
+  actions: {
+    indexUrls: async ({ input, logger, client }) => {
+      const pageUrlsSchema = z.object({
+        urls: z.array(z.string()),
+      })
+      const { urls: pageUrls } = pageUrlsSchema.parse(JSON.parse(input.pageUrls))
+
+      logger.forBot().debug(`Indexing ${pageUrls.length} urls`)
+
+      const scraper = new Scraper(logger, bp.secrets.SCRAPER_API_KEY)
+      let scraperCreditCost = 0
+      const files = await Promise.all(
+        pageUrls.map(async (url) => {
+          const scrapingResponse = await scraper.fetchPageHtml(url)
+          scraperCreditCost += scrapingResponse.cost
+          const response = await client.upsertFile({
+            key: url,
+            size: scrapingResponse.content.byteLength,
+            index: true,
+            contentType: 'text/html',
+          })
+          await axios.put(response.file.uploadUrl, scrapingResponse.content)
+          return response.file.id
+        })
+      )
+
+      return { fileIds: JSON.stringify(files), scraperCreditCost }
```

Comment:
:question:**question:** what is the maximum execution time of an integration? it could fail after running for too long and cause the caller to retry indexing already indexed urls

Reply:
60 seconds